python/lsst/afw/table/base/baseContinued.py

```diff
@@ -127,8 +127,18 @@ def __getitem__(self, key):
                 stop = len(self)
             return self.subset(start, stop, step)
         elif isinstance(key, np.ndarray):
+            if not self.isContiguous():
+                cat = self.__class__(self.table)
+                for rr, ok in zip(self, key):
+                    if ok:
+                        cat.append(rr)
+                return cat
             return self.subset(key)
         elif isinstance(key, Key) or isinstance(key, basestring):
+            if not self.isContiguous():
+                if isinstance(key, basestring):
+                    key = self.schema[key].asKey()
+                return np.array([rr.get(key) for rr in self])
```

Comment:
This is returning a numpy array instead of an afw table. That seems bad.

Reply:
But isn't the whole idea that when you index a `Catalog` with a `Key` you do in fact get a numpy array? Or are you objecting that mutating the returned array would not mutate the `Catalog`?

Reply:
I would appreciate @TallJimbo's view on these changes to afw table too.

Reply:
If you're worried about the mutation problem I mentioned, what if we [made the returned array read only](https://stackoverflow.com/a/5541452/834250)?

Reply:
Oh! I didn't know that these returned a numpy array instead of an afw.Table. That's somewhat surprising to me: I would expect `Thing.__getitem__[slice]` to return another `Thing`.

(thinking out loud here) But I guess `Thing[Key]` gets a column, which is like an array. That's not obvious from the alternative `return self.subset(key)`, which I had trouble finding because it's defined in C++. Maybe we should make the docstring for `__getitem__` more clear about the types of the possible return values? (says the python person who's clearly been staring at too much C++ lately...)

You are correct though that the mutation question is important: `subset` is explicitly a shallow copy, so this would cause a different behavior.